File: Templates/VisualBasic/Model/Person.vb
Imports Xtensive.Orm

Namespace $safeprojectname$
    <HierarchyRoot>
    Public Class MyEntity 
        Inherits Entity
  
        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        <Field, Key>
        Public Property Id As Integer

        <Field>
        Public Property FirstName As String

        <Field>
        Public Property LastName As String

        <Field>
        Public Property Age As Integer

    End Class
End NameSpace

File: Templates/CSharp/Model/Person.cs
using System;
using Xtensive.Orm;

namespace $safeprojectname$
{
	[HierarchyRoot]
	public class Person : Entity
	{
		[Field, Key]
		public int Id { get; private set; }

		[Field]
		public string FirstName { get; set; }

		[Field]
		public string LastName { get; set; }

		[Field]
		public int Age { get; set; }


		public Person()
			: base()
		{
		}

		public Person(Session session)
			: base(session)
		{
		}
	}
}
